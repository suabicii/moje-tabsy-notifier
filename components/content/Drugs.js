import React from "react";
import {Title} from "react-native-paper";
import {View} from "react-native";
import DosingMoments from "./DosingMoments";

function Drugs({drugList}) {
    return (
        <>
            {
                drugList.map(({dosing, name, unit, dosingMoments}, index) =>
                    (
                        <View key={`${name}${index}`}>
                            <Title style={{textAlign: "center"}}>{name} {dosing} {unit} w godz.</Title>
                            <DosingMoments
                                drugName={name}
                                content={Object.entries(dosingMoments)}
                            />
                        </View>
                    )
                )
            }
        </>
    );
}

export default Drugs;