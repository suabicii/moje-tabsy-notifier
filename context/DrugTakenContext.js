import React, {createContext, useContext} from "react";

const DrugTakenContext = createContext([]);

export const useDrugTakenContext = () => useContext(DrugTakenContext);

export default DrugTakenContext;