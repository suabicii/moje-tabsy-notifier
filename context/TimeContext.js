import React, {createContext} from "react";

export const TimeContext = createContext(undefined);