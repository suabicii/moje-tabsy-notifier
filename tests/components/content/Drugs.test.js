import React from "react";
import renderer from "react-test-renderer";
import Drugs from "../../../components/content/Drugs";
import {drugList} from "../../fixtures/drugList";
import DrugTakenContext from "../../../context/DrugTakenContext";
import {TimeContext} from "../../../context/TimeContext";
import MockDate from "mockdate";
import dayjs from "dayjs";

let currentTime;
beforeAll(() => {
    MockDate.set('2020-01-01');
    currentTime = dayjs();
});

const drugTakenChecker = [];
const setDrugTakenChecker = jest.fn();
const setCurrentTime = jest.fn();
const mockUseContext = jest.fn().mockImplementation(() => ({drugTakenChecker, setDrugTakenChecker}));

React.useContext = mockUseContext;

function WrappedComponent() {
    return (
        <TimeContext.Provider value={{currentTime, setCurrentTime}}>
            <DrugTakenContext.Provider value={{drugTakenChecker, setDrugTakenChecker}}>
                <Drugs drugList={drugList}/>
            </DrugTakenContext.Provider>
        </TimeContext.Provider>
    );
}

it('should correctly render drug data', () => {
    const tree = renderer.create(<WrappedComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
});