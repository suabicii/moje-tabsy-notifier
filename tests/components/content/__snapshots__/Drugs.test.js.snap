// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render drug data 1`] = `
[
  <View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
      testID="Xanax"
    >
      Xanax
       
      1
       
      szt.
       w godz.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        – 
        07:00
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#6cc3d5",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "color": "#e8e8e8",
            "elevation": 0,
            "marginLeft": 5,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
          testID="Xanax_hour1"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "flexDirection": "row-reverse",
                },
              ]
            }
          >
            <View
              style={
                {
                  "marginLeft": -4,
                  "marginRight": 12,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.32)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Już zażyłem/-am
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        – 
        18:00
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#6cc3d5",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "color": "#e8e8e8",
            "elevation": 0,
            "marginLeft": 5,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
          testID="Xanax_hour2"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "flexDirection": "row-reverse",
                },
              ]
            }
          >
            <View
              style={
                {
                  "marginLeft": -4,
                  "marginRight": 12,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.32)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Już zażyłem/-am
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
      testID="Witamina C"
    >
      Witamina C
       
      1
       
      szt.
       w godz.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        – 
        12:00
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#6cc3d5",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "color": "#e8e8e8",
            "elevation": 0,
            "marginLeft": 5,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
          testID="Witamina C_hour1"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "flexDirection": "row-reverse",
                },
              ]
            }
          >
            <View
              style={
                {
                  "marginLeft": -4,
                  "marginRight": 12,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.32)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Już zażyłem/-am
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
      testID="Metanabol"
    >
      Metanabol
       
      1
       
      amp. 10 ml
       w godz.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        – 
        12:00
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#6cc3d5",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "color": "#e8e8e8",
            "elevation": 0,
            "marginLeft": 5,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
          testID="Metanabol_hour1"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "flexDirection": "row-reverse",
                },
              ]
            }
          >
            <View
              style={
                {
                  "marginLeft": -4,
                  "marginRight": 12,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.32)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Już zażyłem/-am
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
